src/backend/utils.ts

```diff
@@ -871,6 +875,86 @@ export const spawnAsync = async (
   })
 }
 
+async function ContinueWithFoundWine(
+  selectedWine: string,
+  foundWine: string
+): Promise<{ response: number }> {
+  const { response } = await dialog.showMessageBox({
+    title: i18next.t('box.warning.wine-change.title', 'Wine not found!'),
+    message: i18next.t('box.warning.wine-change.message', {
+      defaultValue:
+        'We could not find the selected wine version to launch this title ({{selectedWine}}). {{newline}} We found another one, do you want to continue launching using {{foundWine}} ?',
+      newline: '\n',
+      selectedWine: selectedWine,
+      foundWine: foundWine
+    }),
+    buttons: [i18next.t('box.yes'), i18next.t('box.no')]
+  })
+
+  return { response }
+}
+
+export async function checkWineBeforeLaunch(
+  appName: string,
+  gameSettings: GameSettings,
+  logFileLocation: string
+): Promise<boolean> {
+  const wineIsValid = await validWine(gameSettings.wineVersion)
+
+  logError(
+    `Wine version ${gameSettings.wineVersion.name} is not valid, trying another one.`,
+    LogPrefix.Backend
+  )
+  appendFileSync(
+    logFileLocation,
+    `Wine version ${gameSettings.wineVersion.name} is not valid, trying another one.`
+  )
+
+  if (!wineIsValid) {
```

Comment:
Could implement an early-return here to avoid one level of indentation
```ts
if (wineIsValid) {
  return true
}
// check if the default wine is valid now
const { wineVersion: defaultwine } = ...
```

Reply:
done